### Welcome to my [README](https://jkmcpherson.github.io/jkmcpherson/)

You can find my publications at [PubMed](https://www.ncbi.nlm.nih.gov/myncbi/1XeOuqmaFjg5z/bibliography/public/), [Google Scholar](https://scholar.google.com/citations?user=MsZPCoIAAAAJ&hl=en), [ResearchGate](https://www.researchgate.net/profile/Jacob-Mcpherson), [SCOPUS](https://www.scopus.com/authid/detail.uri?authorId=57204192355), or [ORCID](https://orcid.org/0000-0001-5486-4945).

[![GitHub followers](https://img.shields.io/github/followers/jkmcpherson.svg?style=social&label=Follow&maxAge=2592000)](https://github.com/jkmcpherson?tab=followers)
[![Twitter Followers](https://badgen.net/twitter/follow/jk_mcpherson)](https://twitter.com/jk_mcpherson)

I keep a [**Netlify Gitbook**](https://jkmcpherson.netlify.app/) where I curate my favorite resources from: 

My Favorite Scientific Journals and Books
---

**Academic and clinical journals** typically starting with the most reliable and relevant search engine, 
 [SCOUPS](https://www.scopus.com/) that usually pulls publications from 
 [*Nature*](https://www.nature.com/), 
 [*Science*](https://www.science.org/), 
 [*Cell*](https://www.cell.com/), 
 [*Nature Medicine*](https://www.nature.com/nm/), 
 [*Science Translational Medicine*](https://www.science.org/journal/stm), 
 [*The Lancet*](https://www.thelancet.com/), 
 [*New England Journal of Medicine* (NEJM)](https://www.nejm.org/), 
 [*Nature Reviews Disease Primers* (NRDP)](https://www.nature.com/nrdp/), and 
 [*Physiological Reviews*](https://journals.physiology.org/journal/physrev)

**Cellular, molecular, and structural biology** in 
[Bruce Albert's 'Molecular Biology of the Cell'](https://brucealberts.ucsf.edu/current-projects/molecular-biology-of-the-cell/)
[*Nature Reviews Molecular Cell Biology*](https://www.nature.com/nrm/), 
[*Nature Structural & Molecular Biology*](https://www.nature.com/nsmb/), 
[*Signal Transduction and Targeted Therapy*](https://www.nature.com/sigtrans/), and
[*Science Signaling*](https://www.science.org/journal/signaling)

**Pharmacology of receptor theory** in 
[Terry Kenakin's 'A Pharmacology Primer'](https://www.sciencedirect.com/book/9780128139578/a-pharmacology-primer), 
[*Pharmacological Reviews*](https://pharmrev.aspetjournals.org/), 
[*Nature Reviews Drug Discovery*](https://www.nature.com/nrd/), the 
[*British Journal of Pharmacology* (BPJ)](https://bpspubs.onlinelibrary.wiley.com/journal/14765381), and the 
[*Journal of Pharmacology and Experimental Therapeutics* (JPET)](https://www.aspet.org/aspet/journals/the-journal-of-pharmacology-and-experimental-therapeutics) 

**Clinical microbiology, metagenomics and microbial ecology (microbiome sciences)** in 
[Manual of Clinical Microbiology](https://www.amazon.com/Clinical-Microbiology-Twelfth-Michael-Pfaller/dp/1555819834)
[Principles of Virology](https://www.amazon.com/Principles-Virology-Multi-ASM-Books/dp/1683670329/)
[*Nature Reviews Microbiology*](https://www.nature.com/nrmicro/), 
[*Clinical Microbiology Reviews*](https://journals.asm.org/journal/cmr), 
[*Journal of Clinical Microbiology* (J Clin Micro)](https://journals.asm.org/journal/jcm), 
[*Nature Microbiology*](https://www.nature.com/nmicrobiol/), 
[*International Society for Microbial Ecology Journal* (ISMEJ)](https://www.nature.com/ismej/), 
[*Cell Host & Microbe*](https://www.cell.com/cell-host-microbe/), the
[*Nature* subject/Microbiology](https://www.nature.com/subjects/microbiology/nature) which includes the more narrow subjects 
[*Nature* subject/Microbiome](https://www.nature.com/subjects/microbiome/nature), 
[*Nature* subject/Communities](https://www.nature.com/subjects/communities/nature), 
[*Nature* subject/Metagenomics](https://www.nature.com/subjects/metagenomics/nature), and the ecology & evolution journals
[*Annual Review of Ecology, Evolution and Systematics*](https://www.annualreviews.org/journal/ecolsys), and 
[*Nature Ecology and Evolution*](https://www.nature.com/natecolevol/)
 
**Infectious Diseases and antimicrobial stewardship pharmacy practice** in 
[*Antimicrobial Agents and Chemotherapy* (AAC)](https://journals.asm.org/journal/aac), 
[*Nature* subject/Antimicrobials](https://www.nature.com/subjects/antimicrobials/nature), 
[*Clinical Infectious Diseases* (CID)](https://academic.oup.com/cid), 
[*The Lancet Infectious Diseases*](https://www.thelancet.com/journals/laninf/home), 
[*NEJM Infectious Diseases*](https://www.nejm.org/infectious-disease), and the CDC's 
[*Morbidity and Mortality Weekly Report* (MMWR)](https://www.cdc.gov/mmwr/) and
[*Emerging Infectious Diseases* (EID)](https://wwwnc.cdc.gov/eid/)

**Nucleic acids research** in 
[Baxevanis, Bader & Wishart's 'Bioinformatics: A Practical Guide to the Analysis of Genes and Proteins'](https://www.amazon.com/Bioinformatics-Practical-Guide-Analysis-Proteins/dp/1119335582/), 
[*Bioinformatics*](https://academic.oup.com/bioinformatics/), 
[*Nature Genetics*](https://www.nature.com/ng/), 
[*Nature Reviews Genetics*](https://www.nature.com/nrg/), 
[*Genome Biology*](https://genomebiology.biomedcentral.com/), 
[*Nucleic Acids Research*](https://academic.oup.com/nar/), the 
[*Annual Review of Genetics*](https://www.annualreviews.org/journal/genet), and the 
[*Annual Review of Genomics and Human Genetics*](https://www.annualreviews.org/journal/genom) 

**Immunology** in 
[*Nature Immunology*](https://www.nature.com/ni/), 
[*Nature Reviews Immunology*](https://www.nature.com/nri/), 
[*Science Immunology*](https://www.science.org/toc/sciimmunol/current), and 
[*Mucosal Immunology*](https://www.nature.com/mi/)

**Laboratory methods** in 
[*Nature Methods*](https://www.nature.com/nmeth/), 
[*Nature Protocols*](https://www.nature.com/nprot/), and 
[Springer|Nature Experiments](https://experiments.springernature.com/)

**Artificial Intelligence** in 
[*Nature Machine Intelligence*](https://www.nature.com/natmachintell/), 
[*Science Robototics*](https://www.science.org/journal/scirobotics), the 
[*IEEE Transactions on Pattern Analysis and Machine Intelligence*](https://ieeexplore.ieee.org/xpl/RecentIssue.jsp?punumber=34),
[*IEEE Transactions on Neural Networks and Learning Systems*](https://ieeexplore.ieee.org/xpl/RecentIssue.jsp?punumber=5962385), the 
[*International Journal of Intelligent Systems*](https://onlinelibrary.wiley.com/journal/1098111x)
[*Information Sciences*](https://www.journals.elsevier.com/information-sciences), the 
[*Physics of Life Review*](https://www.sciencedirect.com/journal/physics-of-life-reviews), 
[*Artificial Intelligence Review*](https://www.springer.com/journal/10462), 
[*Knowledge-Based Systems*](https://www.journals.elsevier.com/knowledge-based-systems), 
[*Neural Networks*](https://www.journals.elsevier.com/neural-networks),
[*Neural Computing and Applications*](https://www.springer.com/journal/521), the 
[*International Journal of Computer Vision*](https://www.springer.com/journal/11263), and the journal
[*Pattern Recognition*](https://www.sciencedirect.com/journal/pattern-recognition)

[**Microscopy, Chemistry, and Structural Biology**]()
---

The [UHCOP Core](https://uh.edu/pharmacy/research/core-lab/) houses a [Confocal](https://www.leica-microsystems.com/products/confocal-microscopes/p/stellaris-sted/), [NMR](https://www.jeolusa.com/PRODUCTS/Nuclear-Magnetic-Resonance/JNM-ECZL-series-FT-NMR), [LCMS](https://sciex.com/products/mass-spectrometers/triple-quad-systems/triple-quad-5500-system), [ddPCR](https://www.bio-rad.com/en-us/life-science/digital-pcr/qx200-ddpcr-system-with-automated-droplet-generator), [chromatography](https://www.bionity.com/en/products/128437/aekta-pure-flexible-andintuitive-protein-purification-system.html), and a [biolayer interferometer](https://www.sartorius.com/en/products/protein-analysis/octet-bli-detection/octet-label-free-detection-systems)

[**Confocal fluorescence microscopy**]() including the [Leica learning resources](https://www.leica-microsystems.com/science-lab/science-lab-home)

[**Cryogenic electronic microscopy (CryoEM)**]() that [Yale CryoEM](https://cryoem.yale.edu/cryo-em/workshops-and-online-courses) and [UNC CryoEM Core](https://www.med.unc.edu/cryo-em/) lists [resources available](https://www.med.unc.edu/cryo-em/cryoem-links-and-resources/), such as those by [Grant Jensen, CalTech](https://cryo-em-course.caltech.edu/), the [NIGMS National Resource for Automated Molecular Microscopy (NRAMM)](https://nramm.nysbc.org/workshops-and-courses/), and [Thermo Fisher's EM-Learning](https://em-learning.com/) and [CryoEM Learning Center](https://www.thermofisher.com/us/en/home/electron-microscopy/life-sciences/learning-center.html)

Nuclear Magnetic Resonance Imaging (NMR)

Liquid Chromatography Mass Spectrometry (LCMS)

[**Structural Biology**]()
---
The [Theoretical and Computational Biophysics Group (TCBG)]() provides [tutorial-based training](https://www.ks.uiuc.edu/Training/Tutorials/) and a database of software for 3D molecular [building](https://www.ks.uiuc.edu/Development/biosoftdb/biosoft.cgi?&category=3), [dynamics](https://www.ks.uiuc.edu/Development/biosoftdb/biosoft.cgi?&category=2) and [visualization](https://www.ks.uiuc.edu/Development/biosoftdb/biosoft.cgi?&category=1)

Further resources can be found notably at the [EMBLI-EBI](https://www.ebi.ac.uk/training/search-results?query=structural-biology), Stanford

[**Scientific Programming: Bioinformatics & Computational Biology**]()
---
  
Genomics, metagenomics, transcriptomics, proteomics, and metabolomics [in Python](https://www.python.org/) using the [BioPython](https://biopython.org/) library

[Swiss Institute for Bioinformatics (SIB)](https://www.sib.swiss/)
    
[National Center for Biotechnology and Information (NCBI)](https://ncbiinsights.ncbi.nlm.nih.gov/)
  
[The Bacterial and Viral Bioinformatics Resource Center (BVBRC)](https://www.bv-brc.org/)
    
[European Molecular Biology Laboratories (EMBL)](https://www.embl.org/) [European Bioinformatics Institute (EBI)](https://www.ebi.ac.uk/research)
  
The commercial [QIAGEN Digital Insights](https://digitalinsights.qiagen.com/) and [Schrodinger](https://www.schrodinger.com/)
  
The microbial [QIIME2](https://qiime2.org/), [PICRUST2](https://github.com/picrust/picrust2), and the [interactive Human Microbiome Project (iHMP)](https://portal.hmpdacc.org/)

[**Statistical programming and data visualization (in R)**]()
---
see the statistics primers towards the bottom of the *Nature* collection [*Statistics for Biologists*](https://www.nature.com/collections/qghhqm) and the sub-page [*Points of Significance*](https://www.nature.com/collections/qghhqm/pointsofsignificance)

[Antoine Swoeney's](https://antoinesoetewey.com/) excellent [Blogdown](https://github.com/rstudio/blogdown) titled ['Stats and R'](https://github.com/AntoineSoetewey/statsandr) that I recommend starting with ['What statistical test should I do?'](https://statsandr.com/blog/what-statistical-test-should-i-do/) and clicking the in-picture end-node links. 

The [*Handbook of Statistical Analyses Using R* (HSAUR)](https://rdrr.io/cran/HSAUR/)


[**Artificial intelligence, machine learning, and deep learning (in Python, R)**]()
---


[**Computer science communities:**]()
---
[R](https://www.r-project.org/), [RStudio](https://www.rstudio.com/), [Tidyverse](https://www.tidyverse.org/), [Bookdown](https://bookdown.org/), [Python](https://www.python.org/), [Anaconda](https://www.anaconda.com/) , [Julia](https://julialang.org/), [Docker](https://www.docker.com/), [Git](https://git-scm.com/), [LaTeX](https://www.latex-project.org/), [VSCode](https://code.visualstudio.com/), [AWS](https://aws.amazon.com/), [Ubuntu](https://www.ubuntu.com)


### Languages, Tools, and Skills 🛠
---
[![Docker](https://img.shields.io/badge/docker-%230db7ed.svg?style=for-the-badge&logo=docker&logoColor=white)](https://www.docker.com/)
[![Git](https://img.shields.io/badge/git-%23F05033.svg?style=for-the-badge&logo=git&logoColor=white)](https://git-scm.com/)
[![GitHub](https://img.shields.io/badge/GitHub-100000?style=for-the-badge&logo=github&logoColor=white)](https://github.com/)
[![VSCode](https://img.shields.io/badge/vs%20code-007ACC?style=for-the-badge&logo=visual%20studio%20code&logoColor=white)](https://code.visualstudio.com/)
[![Ubuntu](https://img.shields.io/badge/Ubuntu-E95420?style=for-the-badge&logo=ubuntu&logoColor=white)](https://www.ubuntu.com)
[![AWS](https://img.shields.io/badge/AWS-%23FF9900.svg?style=for-the-badge&logo=amazon-aws&logoColor=white)](https://aws.amazon.com/)
[![R](https://img.shields.io/badge/r-%23276DC3.svg?style=for-the-badge&logo=r&logoColor=white)](https://www.r-project.org/)
[![Markdown](https://img.shields.io/badge/markdown-%23000000.svg?style=for-the-badge&logo=markdown&logoColor=white)](https://www.markdownguide.org/)
[![LaTeX](https://img.shields.io/badge/latex-%23008080.svg?style=for-the-badge&logo=latex&logoColor=white)](https://www.latex-project.org/)
[![Python](https://img.shields.io/badge/python-3670A0?style=for-the-badge&logo=python&logoColor=ffdd54)](https://www.python.org/)
[![SciPy](https://img.shields.io/badge/SciPy-%230C55A5.svg?style=for-the-badge&logo=scipy&logoColor=%white)](https://scipy.org/)
[![NumPy](https://img.shields.io/badge/numpy-%23013243.svg?style=for-the-badge&logo=numpy&logoColor=white)](https://numpy.org)
[![Pandas](https://img.shields.io/badge/pandas-%23150458.svg?style=for-the-badge&logo=pandas&logoColor=white)](https://pandas.pydata.org)
[![Plotly](https://img.shields.io/badge/Plotly-%233F4F75.svg?style=for-the-badge&logo=plotly&logoColor=white)](https://plotly.com)
[![Keras](https://img.shields.io/badge/Keras-%23D00000.svg?style=for-the-badge&logo=Keras&logoColor=white)](https://keras.io)
[![PyTorch](https://img.shields.io/badge/PyTorch-%23EE4C2C.svg?style=for-the-badge&logo=PyTorch&logoColor=white)](https://pytorch.org)
[![TensorFlow](https://img.shields.io/badge/TensorFlow-%23FF6F00.svg?style=for-the-badge&logo=TensorFlow&logoColor=white)](https://www.tensorflow.org)
[![Julia](https://img.shields.io/badge/-Julia-9558B2?style=for-the-badge&logo=julia&logoColor=white)](https://julialang.org/)
[![Netlify](https://img.shields.io/badge/netlify-%23000000.svg?style=for-the-badge&logo=netlify&logoColor=#00C7B7)](https://www.netlify.com/)
[![Made withJupyter](https://img.shields.io/badge/Made%20with-Jupyter-orange?style=for-the-badge&logo=Jupyter)](https://jupyter.org/try)
[![Binder](https://mybinder.org/badge_logo.svg)](https://mybinder.org/)

[![Contribution Stats](https://github-contribution-stats.vercel.app/api/?username=jkmcpherson)](https://github.com/LordDashMe/github-contribution-stats/)
[![Top Langs](https://github-readme-stats.vercel.app/api/top-langs/?username=jkmcpherson&layout=default)](https://github.com/anuraghazra/github-readme-stats)

<!-- badges: start -->
[![CRAN_Status_Badge](https://www.r-pkg.org/badges/version/ggplot2)](https://cran.r-project.org/package=ggplot2)
[![R-CMD-check](https://github.com/tidyverse/ggplot2/workflows/R-CMD-check/badge.svg)](https://github.com/tidyverse/ggplot2/actions)
[![Codecov test
coverage](https://codecov.io/gh/tidyverse/ggplot2/branch/main/graph/badge.svg)](https://app.codecov.io/gh/tidyverse/ggplot2?branch=main)
<!-- badges: end -->
<a href="https://github.com/rstudio/cheatsheets/blob/master/data-visualization.pdf"><img src="https://raw.githubusercontent.com/rstudio/cheatsheets/master/pngs/thumbnails/data-visualization-cheatsheet-thumbs.png" width="630" height="252"/></a>

<!-- badges: start -->
[![CRAN status](https://www.r-pkg.org/badges/version/dplyr)](https://cran.r-project.org/package=purrr)
[![R-CMD-check](https://github.com/tidyverse/purrr/actions/workflows/R-CMD-check.yaml/badge.svg)](https://github.com/tidyverse/purrr/actions/workflows/R-CMD-check.yaml)
[![Codecov test coverage](https://codecov.io/gh/tidyverse/purrr/branch/main/graph/badge.svg)](https://app.codecov.io/gh/tidyverse/purrr?branch=main)
<!-- badges: end -->
<a href="https://github.com/rstudio/cheatsheets/blob/main/purrr.pdf"><img src="https://raw.githubusercontent.com/rstudio/cheatsheets/main/pngs/thumbnails/purrr-cheatsheet-thumbs.png" width="630" height="252"/></a>


[![GitHub forks](https://img.shields.io/github/forks/jkmcpherson/jkmcpherson?style=social&label=Fork&maxAge=2592000)](https://GitHub.com/jkmcpherson/jkmcpherson/)
[![GitHub stars](https://img.shields.io/github/stars/jkmcpherson/jkmcpherson?style=social&label=Star&maxAge=2592000)](https://GitHub.com/jkmcpherson/jkmcpherson/)
[![DOI:10.1007/978-3-319-76207-4_15](https://zenodo.org/badge/DOI/10.1007/978-3-319-76207-4_15.svg)](https://doi.org/10.1007/978-3-319-76207-4_15)
[![Citation Badge](https://api.juleskreuer.eu/citation-badge.php?doi=10.1126/science.1058040)](https://juleskreuer.eu/projekte/citation-badge/)

[![github-readme-twitter](https://github-readme-twitter.gazf.vercel.app/api?id=nature)](https://twitter.com/Nature) 
[![github-readme-twitter](https://github-readme-twitter.gazf.vercel.app/api?id=thelancet)](https://twitter.com/TheLancet) 

This is an example of a [**Bookdown**](https://github.com/rstudio/bookdown) book based on [**R Markdown**](https://github.com/rstudio/rmarkdown), created by by [*Yi Hui*](https://bookdown.org/yihui/bookdown/) with [*reference documentation*](https://pkgs.rstudio.com/bookdown)
